SP5WWP/m17-decoder/m17-decoder-sym.c

```diff
@@ -551,7 +549,7 @@ int main(int argc, char* argv[])
                     //The Signature is not encrypted
                     
                     //AES
-                    if(encryption==ENCR_AES && fn<0x7FFC)
+                    if(encryption==ENCR_AES && (fn % 0x8000)<0x7FFC)
```

Comment:
in m17-decoder-sym.c, the fn value also includes the EOT bit, without this, the last frame on a non-signed stream was not being decrypted properly since it would have a value of 0x8000 or higher. Same thing was happening on scrambler and AES.

Without Sig:

```
./m17-coder-sym -D -k b869 > ../m17-decoder/float.sym
Scrambler key: 0xB869 (16-bit)

tail -f float.sym | ./m17-decoder-sym -k b869
...
DST: FFFFFFFFFFFF SRC: 00004B13D106 TYPE: 002D (STREAM: VOICE, ENCR: SCRAM 16-bit, CAN: 0) 
FN: 0036 PLD: 69696969696969696969696969696969
FN: 0037 PLD: 69696969696969696969696969696969
FN: 0038 PLD: 69696969696969696969696969696969
FN: 0039 PLD: 69696969696969696969696969696969
FN: 003A PLD: 69696969696969696969696969696969
FN: 003B PLD: 69696969696969696969696969696969
DST: FFFFFFFFFFFF SRC: 00004B13D106 TYPE: 002D (STREAM: VOICE, ENCR: SCRAM 16-bit, CAN: 0) 
FN: 803C PLD: 69696969696969696969696969696969

```

With Sig:
```
./m17-coder-sym -D -k b869 -s sample_priv_key.txt > ../m17-decoder/float.sym
Scrambler key: 0xB869 (16-bit)
Signature: 84029EB325937655578D0CD730509930
           178EE99E2E5BDCEC928EC8F2E6841C78
           54B78713627278418BDC635C45AAF08D
           C7908D62B6EA94FF29168009A190F946

tail -f float.sym | ./m17-decoder-sym -k b869 -s sample_pub_key.txt

...
DST: FFFFFFFFFFFF SRC: 00004B13D106 TYPE: 082D (STREAM: VOICE, ENCR: SCRAM 16-bit, CAN: 0, SIGNED) 
FN: 003C PLD: 69696969696969696969696969696969
FN: 7FFC PLD: 84029EB325937655578D0CD730509930
FN: 7FFD PLD: 178EE99E2E5BDCEC928EC8F2E6841C78
FN: 7FFE PLD: 54B78713627278418BDC635C45AAF08D
FN: FFFF PLD: C7908D62B6EA94FF29168009A190F946
Signature OK

```

Reply:
> This line and line 561 need to check if signatures are enabled.

Depends, are frame numbers 7FFC, 7FFD, 7FFE, and 7FFF exclusively for Signatures, or are we still allowed to use them on non-stream frames, assumming somebody can actually talk for 20+ minutes straight to get that high.

Reply:
These values are perfectly valid for non-signed streams.

Reply:
I'll make some tweaks to check for signed_str